### PR TITLE
Inspect linkages

### PR DESCRIPTION
**Pull request by ocefpaf**
Run inspect linkages and refresh patch to work with `conda-forge`'s `glib`.

### CODE REVIEW

File: recipe/site_specific.patch [hunk 1]
```diff
@@ -104,7 +104,7 @@ diff -Naur PyFerret-67aed0fa29f2d3f28e624030d746d23d146fc387.orig/platform_speci
 -			  -I/usr/lib64/glib-2.0/include \
 +			  -I$(PREFIX)/include/pango-1.0 \
 +			  -I$(PREFIX)/include/glib-2.0 \
-+			  -I$(PREFIX)/lib64/glib-2.0/include \
```
Comment (ocefpaf) [inline recipe/site_specific.patch]: @msarahan just to let you know that `defaults` version of `glib` has its libraries on `lib64` instead of `lib`.

**Comment by ocefpaf** [comment]
@msarahan did the syntaxe for conda-inspect linkages changed?

``` shell
+ /bin/bash -x -e /feedstock_root/build_artefacts/test-tmp_dir/run_test.sh
+ conda inspect linkages -n _test pyferret
usage: conda-inspect [-h] [-c] [-f] [-q] [--no-remove] [dir [dir ...]]
conda-inspect: error: unrecognized arguments: -n _test pyferret
TESTS FAILED: pyferret-7.0-np110_4

./ci_support/run_docker_build.sh returned exit code 1
```

**Comment by msarahan** [comment]
No, those arguments should be added in https://github.com/conda/conda-build/blob/1.21.x/conda_build/main_inspect.py#L89

Moreover, main_inspect hasn't changed since June 14, so I'm not sure what has happened here.  It is also concerning that it seems to not be showing any of the `linkages` arguments:

https://github.com/conda/conda-build/blob/1.21.x/conda_build/main_inspect.py#L65-L88

Is this a different conda version than usual?  Conda-build uses conda's argparse to hook into.  I don't think this has changed in conda, but it is a possibility.

**Comment by msarahan** [comment]
I think this is something wrong with the conda-build recipe on conda-forge.  That is the main change here, AFAICT.  I'll see about adding a test and fixing it.

**Comment by ocefpaf** [comment]
> I think this is something wrong with the conda-build recipe on conda-forge.  That is the main change here, AFAICT.  I'll see about adding a test and fixing it.

Thanks @msarahan. I think we should pull the packages as this will break all the builds that use `inspect`.

**Comment by msarahan** [comment]
SGTM.  You want to do that, or should I?

**Comment by ocefpaf** [comment]
> SGTM. You want to do that, or should I?

I was just waiting for your opinion. Doing it now. (And I see you are already on your way to solve it in the feedstock. Thanks!)

**Comment by ocefpaf** [comment]
@msarahan just to let you know pulling `conda-build` from `conda-forge` and re-starting this did the trick. I won't merge until we have https://github.com/conda/conda-build/pull/1165 (or https://github.com/conda-forge/conda-build-feedstock/pull/2 ?) so we can test the new `conda-build` package here. Thanks!

**Comment by msarahan** [comment]
Cool.  It's looking good based on https://github.com/conda-forge/conda-build-feedstock/pull/2.  I'm waiting on the CI's, then will tag 1.21.10 and update the CF feedstock recipe.

**Comment by msarahan** [comment]
The plan right now is that @ilanschnell will patch the conda-build setup.py to do a conda-build package release (https://github.com/conda/conda/issues/3236#issuecomment-237953770).  I have asked him to hold off until 1.21.11 is released - probably tomorrow, after https://github.com/conda/conda-build/issues/1167#issuecomment-238004121 is settled.

Looking forward, at this point, I don't think conda-forge will be able to build a fully-working conda-build until https://github.com/conda/conda/pull/3224 is merged, or until there's another fix for why these subcommand entry points are not working right.

**Comment by ocefpaf** [comment]
Thanks for the feedback @msarahan!

**Pull request merged**